### PR TITLE
Improve postprocessing efficiency

### PR DESCRIPTION
**Pull request by kevinhu**
This PR makes a couple of small refactors to improve performance with processing logits. On an L40S, this translates to a ~3x speed improvement.

1. In `get_token_spans`, `getattr` calls to `tokenizer.cls_token, tokenizer.sep_token, tokenizer.pad_token` dominate and are replaced with a one-time definition when the tokenizer is first instantiated.
2. In `token_to_char_probs`, assignment of `char_probs` is replaced with vectorized indexing instead of a for loop.

Traces from pyinstrument:

Before:
Screenshot 2025-06-06 at 16 51 44

After:
Screenshot 2025-06-06 at 16 50 03

### CODE REVIEW

**Comment by markus583** [comment]
Thanks for the suggestions, looks all good to me! I will merge it.
